### DISCUSSION TITLE
GCS / S3 backend support

### DISCUSSION THREAD

**Discussion by mnsRG** [oxigraph/oxigraph#928] [Ideas]
Hi there, is there any plan to support cloud storage backends (like GCS or S3) ? I am currently using GCS as a backend for a cloud application that uses Oxigraph, by utilising GCS Fuse (mounting a GCS bucket as a volume to my Cloud Run instance), but there are many problems that pop up (including, but not limited to, file locking etc). Other projects, like KuzuDB (graph database) or LanceDB (vector database) use `gs://` connection to handle persistence in such cases and it seems to work pretty stable. 
Would that be something for Oxigraph too?
Thanks

**Comment by Tpt** [comment]
Thank you for usingOxigraph even in this kind of painful setting.

This is a great question. Currently I have no concrete short/middle term plans to support cloud storage backends, sadly. The storage system Oxigraph currently uses, RocksDB, has no support for cloud storage natively. However, there is [rocksdb-cloud](https://github.com/rockset/rocksdb-cloud) that seems to have some support for AWS but sadly no GCS. Using it instead of upstream RocksDB might not be too hard. If I have some free time I might investigate that (but if you want to do it, it would be very welcome).

Also, I have some long term plan to migrate from RocksDB to something else (not sure what yet). Cloud storage is definitely something I should take into account when making a possible switch. Thank you for raising this!

**Reply by mnsRG** [reply]
Thanks for your response! It also seems that there is an [open pull request](https://github.com/rockset/rocksdb-cloud/pull/300) on RocksDB-Cloud for supporting GCS backend. In general considering cloud storage would be awesome as it will help to bring Oxigraph to production in serverless setups :)  🙏
